cmd/ftl/cmd_serve.go

```diff
@@ -41,14 +39,27 @@ func (s *serveCmd) Run(ctx context.Context) error {
 
 	wg, ctx := errgroup.WithContext(ctx)
 
+	basePort, err := strconv.Atoi(s.Bind.Port())
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	portAllocator := internal.NewPortAllocator(int32(basePort)) //nolint:gosec
```

Comment:
I think we should call this `NewBindAllocator()` and also pass in the bind address (`127.0.0.1` in this case) so that it can be reused elsewhere.

It also (my fault) shouldn't be `internal`, because it's being used by a function that is public. ie. a private type is exposed on a public function signature.